blog/models.py

```diff
@@ -3,28 +3,63 @@
 from account.models import User
 from problem.models import Problem
 from django.utils.translation import ugettext_lazy as _
+from submission.models import Submission
+from contest.models import Contest
+from contest.models import ContestProblem
+
+
+class BlogManager(models.Manager):
+
+    def get_status_list(self, show_all=False, filter_user=None):
+        q = models.Q()
+        if not show_all:
+            q &= models.Q(is_reward=True, contest=None)
+            if (filter_user):
+                q |= models.Q(contest__participants__username__contains=filter_user.username)
+                q |= models.Q(contest__managers__username__contains=filter_user.username)
```

Comment:
I think `contest__managers=filter_user` is what you want. Also plz check the SQL expression expanded from this query. (in Django debug toolbar)